packages/app-check/src/index.ts

```diff
@@ -48,8 +53,8 @@ function registerAppCheck(firebase: _FirebaseNamespace): void {
       ComponentType.PUBLIC
     )
       .setServiceProps({
-        ReCaptchaV3Provider,
-        CustomProvider
+        ReCaptchaV3Provider: ReCaptchaV3ProviderImpl,
+        CustomProvider: CustomProviderImpl
```

Comment:
Should use `typeof` because we want the constructor type, not the instance type.

Reply:
Did you mean to add `typeof` here or in the `declare` block below? I added it to the `declare` block because it's not a type here but let me know if I'm wrong.

Reply:
Sorry, wrong place. I meant in the `declare` block.